src/main/kotlin/org/rust/ide/template/macros/RustSuggestIndexNameMacro.kt

```diff
@@ -0,0 +1,38 @@
+package org.rust.ide.template.macros
+
+import com.intellij.codeInsight.template.*
+import com.intellij.psi.PsiDocumentManager
+import org.rust.ide.template.RustContextType
+
+class RustSuggestIndexNameMacro : Macro() {
+    override fun getName(): String = "rustSuggestIndexName"
+    override fun getPresentableName(): String = "rustSuggestIndexName()"
+
+    override fun getDefaultValue(): String = "a"
+
+    override fun calculateResult(params: Array<out Expression>, context: ExpressionContext): Result? {
+        if (params.isNotEmpty()) return null
+
+        val project = context.project
+        val offset = context.startOffset
+        val document = context.editor?.document ?: return null
+        val file = PsiDocumentManager.getInstance(project).getPsiFile(document) ?: return null
+        val place = file.findElementAt(offset)
+        val pats = getPatBindingsVisibleAt(place)
+
+        chooseLetterLoop@ for (letter in 'i'..'z') {
+            for (pat in pats) {
```

Comment:
yeah, planned to do manual checks if pattern is defined after place but it's not necessary